lib/middleware/routes-metadata/routes-metadata.js

```diff
@@ -306,12 +306,12 @@ const pageHandler = async (req, res) => {
     }
 
     const componentInstances = jp.query(pageInstance, '$..[?(@._type)]')
-    componentInstances.forEach(componentInstance => {
+    for await (let componentInstance of componentInstances) {
       const componentInstanceController = getInstanceController(componentInstance)
       if (componentInstanceController.preUpdateContents) {
-        componentInstance = componentInstanceController.preUpdateContents(componentInstance, userData, pageInstance)
+        componentInstance = await componentInstanceController.preUpdateContents(componentInstance, userData, pageInstance)
```

Comment:
Can remove unused assignment of componentInstance

Reply:
Removed